tendrl/alerting/notification/handlers/mail_handler.py

```diff
@@ -0,0 +1,276 @@
+                    'receive all alerts'
+                )
+            if config.get('clusters') is None:
+                raise InvalidHandlerConfig(
+                    "'clusters' is manddatory attribute for non admin users"
+                    "The possible values could be '*' or list of cluster ids"
+                )
+        else:
+            if config.get('email_smtp_server') is None:
+                raise InvalidHandlerConfig(
+                    'email_smtp_server is a mandatory field'
+                )
+            if config['auth'] is '':
+                return True
+            else:
+                if 'email_pass' not in config:
+                    raise InvalidHandlerConfig(
+                        'email_pass is a mandatory field if auth is chosen'
+                    )
+        return True
+
+    def add_destination(self, conf):
+        config = ast.literal_eval(conf)
+        try:
+            if self.validate_config(config):
+                self.etcd_server.client.write(
+                    '/alerting/notification_medium/email/config/%s' % config[
+                        'email_id'],
+                    config
+                )
+        except (
+            NotificationDispatchError,
+            InvalidHandlerConfig,
+            EtcdConnectionFailed,
+            EtcdNotDir,
+            KeyError,
+            SyntaxError,
+            ValueError
+        ) as ex:
+            LOG.error(
+                'Failed to add email config %s. Error %s' % (conf, str(ex))
+            )
+            raise NotificationDispatchError(str(ex))
+
+    def set_destinations(self):
+        admin_config = {}
+        user_configs = []
+        try:
+            configs = self.etcd_server.client.read(
+                '/alerting/notification_medium/email/config',
+                recursive=True
+            )
+            for config in configs._children:
+                mail_config = ast.literal_eval(config['value'])
+                if mail_config['is_admin']:
+                    if mail_config.get('auth') is None:
+                        mail_config['auth'] = ""
+                    admin_config = mail_config
+                else:
+                    user_configs.append(mail_config)
+            self.admin_config = admin_config
+            self.user_configs = user_configs
+        except EtcdKeyNotFound as ex:
+            LOG.error(
+                'Could not fetch mail configuration.Error %s'
+                % str(ex)
+            )
+        except (
+            EtcdConnectionFailed,
+            EtcdNotDir,
+            ValueError,
+            KeyError,
+            SyntaxError
+        ) as ex:
+            LOG.error(
+                'Could not fetch mail configuration.Error %s'
+                % str(ex)
+            )
+            raise NotificationDispatchError(str(ex))
+
+    def format_message(self, alert):
+        return "Subject: [Alert] %s, %s threshold breached\n\n%s" % (
```

Comment:
No I don't think this needs to be configurable at-least that's not planned as of now